tools/engine_tool/lib/src/build_plan.dart

```diff
@@ -84,14 +90,48 @@ final class BuildPlan {
     required this.useRbe,
     required this.useLto,
     required this.concurrency,
-  }) {
+    required Iterable<String> extraGnArgs,
+  }) : extaGnArgs = List.unmodifiable(extraGnArgs) {
     if (!useRbe && strategy == BuildStrategy.remote) {
       throw FatalError(
         'Cannot use remote builds without RBE enabled.\n\n$_rbeInstructions',
       );
     }
   }
 
+  /// Arguments that cannot be provided to [BuildPlan.extraGnArgs].
+  ///
+  /// Instead, provide them explicitly as other [BuildPlan] arguments.
+  @visibleForTesting
+  static const reservedGnArgs = {
+    _flagRbe,
+    _flagLto,
+    'no-$_flagRbe',
+    'no-$_flagLto',
+  };
+
+  static void _checkExtraGnArgs(Iterable<String> gnArgs) {
+    for (final arg in gnArgs) {
+      if (!arg.startsWith('--') || arg.contains('=') || arg.contains(' ')) {
+        throw FatalError(
+          'Arguments provided to --gn-args must be flags (booleans) and be '
+          'specified as either in the format "--flag". Options that are not '
```

Comment:
Did you mean to say either --flag or --no-flag here?

Reply:
Yup! Thanks, done!